### PR TITLE
POL-1297 Azure Security Policy Revamps: Part 3

### PR DESCRIPTION
**Pull request by XOmniverse**
### Description

This is a revamp of several Azure Security policies. Please see their respective CHANGELOGs and READMEs for details.

This also fixes a small issue in the Azure Rightsize SQL policy with how actions are logged.

### Link to Example Applied Policy

https://app.flexera.com/orgs/6/automation/applied-policies/projects/7954?policyId=66a3e4d9a09cd94832037277
https://app.flexera.com/orgs/6/automation/applied-policies/projects/7954?policyId=66a3e4f4c13fd765a2f75484
https://app.flexera.com/orgs/6/automation/applied-policies/projects/7954?policyId=66a3e513a09cd94832037278
https://app.flexera.com/orgs/6/automation/applied-policies/projects/7954?policyId=66a3e587a09cd94832037279
https://app.flexera.com/orgs/6/automation/applied-policies/projects/7954?policyId=66a3e59ac13fd765a2f75485
https://app.flexera.com/orgs/6/automation/applied-policies/projects/7954?policyId=66a3e5b1a09cd9483203727a
https://app.flexera.com/orgs/6/automation/applied-policies/projects/7954?policyId=66a3e602a09cd9483203727b
https://app.flexera.com/orgs/6/automation/applied-policies/projects/7954?policyId=66a3e614a09cd9483203727c
https://app.flexera.com/orgs/6/automation/applied-policies/projects/7954?policyId=66a3e62fc13fd765a2f75486
https://app.flexera.com/orgs/6/automation/applied-policies/projects/7954?policyId=66a3e641c13fd765a2f75487

### Contribution Check List

- [X] New functionality includes testing.
- [X] New functionality has been documented in the README if applicable
- [X] New functionality has been documented in CHANGELOG.MD

### CODE REVIEW

File: security/azure/sql_ad_admin/README.md [hunk 1]
```diff
@@ -1,39 +1,46 @@
 # Azure Ensure SQL Server AD Admin Configured
```
Comment (nia-vf1) [inline security/azure/sql_ad_admin/README.md:1]:
```suggestion
# Azure SQL Servers Without Active Directory Admin
```

File: security/azure/sql_publicly_accessible_managed_instance/README.md [hunk 2]
```diff
@@ -1,46 +1,55 @@
 # Azure Publicly Accessible SQL Managed Instances
```
Comment (nia-vf1) [inline security/azure/sql_publicly_accessible_managed_instance/README.md:1]:
```suggestion
# Azure Publicly-Accessible SQL Managed Instances
```

File: security/azure/sql_auditing_retention/sql_auditing_retention.pt [hunk 3]
```diff
@@ -1,13 +1,13 @@
-name "Azure Ensure SQL Server Minimum Auditing Retention Of 90 Days"
+name "Azure SQL Servers With Insufficient Auditing Retention"
 rs_pt_ver 20180301
 type "policy"
-short_description "Report if any SQL server instances do not have auditing retention configured for 90 days or more. \n See the [README](https://github.com/flexera-public/policy_templates/tree/master/security/azure/sql_auditing_retention) and [docs.rightscale.com/policies](https://docs.rightscale.com/policies/) to learn more."
+short_description "Report any SQL server instances that do not have auditing retention configured for a user-specified number of days. See the [README](https://github.com/flexera-public/policy_templates/tree/master/security/azure/sql_auditing_retention) and [docs.rightscale.com/policies](https://docs.rightscale.com/policies/) to learn more."
```
Comment (nia-vf1) [inline security/azure/sql_auditing_retention/sql_auditing_retention.pt:4]:
```suggestion
short_description "Reports any SQL server instances that do not have auditing retention configured for a user-specified number of days. See the [README](https://github.com/flexera-public/policy_templates/tree/master/security/azure/sql_auditing_retention) and [docs.rightscale.com/policies](https://docs.rightscale.com/policies/) to learn more."
```

File: security/azure/sql_server_va/sql_server_va.pt [hunk 4]
```diff
@@ -1,13 +1,13 @@
-name "Azure Ensure SQL Server Vulnerability Assessment (VA) Enabled"
+name "Azure SQL Servers Without Vulnerability Assessment (VA) Enabled"
 rs_pt_ver 20180301
 type "policy"
-short_description "Report if any SQL server instances do not have Vulnerability Assessment (VA) enabled. \n See the [README](https://github.com/flexera-public/policy_templates/tree/master/security/azure/sql_server_va) and [docs.rightscale.com/policies](https://docs.rightscale.com/policies/) to learn more."
+short_description "Report any SQL server instances that do not have Vulnerability Assessment (VA) enabled. See the [README](https://github.com/flexera-public/policy_templates/tree/master/security/azure/sql_server_va) and [docs.rightscale.com/policies](https://docs.rightscale.com/policies/) to learn more."
```
Comment (nia-vf1) [inline security/azure/sql_server_va/sql_server_va.pt:4]:
```suggestion
short_description "Reports any SQL server instances that do not have Vulnerability Assessment (VA) enabled. See the [README](https://github.com/flexera-public/policy_templates/tree/master/security/azure/sql_server_va) and [docs.rightscale.com/policies](https://docs.rightscale.com/policies/) to learn more."
```

File: security/azure/sql_server_va_admins/sql_server_va_admins.pt [hunk 5]
```diff
@@ -1,13 +1,13 @@
-name "Azure Ensure SQL Server VA Notify Admins/Subscription Owners"
+name "Azure SQL Servers Vulnerability Assessment Does Not Notify Admins"
 rs_pt_ver 20180301
 type "policy"
-short_description "Report if any SQL server instances are not configured in VA to also notify admins and subscription owners. \n See the [README](https://github.com/flexera-public/policy_templates/tree/master/security/azure/sql_server_va_admins) and [docs.rightscale.com/policies](https://docs.rightscale.com/policies/) to learn more."
+short_description "Report any SQL server instances that are not configured in Vulnerability Assessment (VA) to notify admins and subscription owners. See the [README](https://github.com/flexera-public/policy_templates/tree/master/security/azure/sql_server_va_admins) and [docs.rightscale.com/policies](https://docs.rightscale.com/policies/) to learn more."
```
Comment (nia-vf1) [inline security/azure/sql_server_va_admins/sql_server_va_admins.pt:4]:
```suggestion
short_description "Reports any SQL server instances that are not configured in Vulnerability Assessment (VA) to notify admins and subscription owners. See the [README](https://github.com/flexera-public/policy_templates/tree/master/security/azure/sql_server_va_admins) and [docs.rightscale.com/policies](https://docs.rightscale.com/policies/) to learn more."
```

Comment (nia-vf1) [reply security/azure/sql_server_va_admins/sql_server_va_admins.pt:4]: Not sure whether the README's description is correct or the template's one is but they read slightly differently

File: security/azure/sql_server_va_emails/sql_server_va_emails.pt [hunk 6]
```diff
@@ -126,37 +329,77 @@ datasource "ds_sql_servers_with_va" do
...
-
-script "js_bad_sql_servers", type: "javascript" do
-  parameters "ds_sql_servers_with_va"
+script "js_bad_azure_sql_servers", type: "javascript" do
+  parameters "ds_sql_servers_with_va", "ds_applied_policy"
   result "result"
   code <<-EOS
-  result = []
+  bad_servers = _.filter(ds_sql_servers_with_va, function(db) {
+    return db['vaEmailList'] == '' || db['vaEmailList'] == null || db['vaEmailList'] == undefined || db['vaEmailList'].length == 0
```
Comment (nia-vf1) [inline security/azure/sql_server_va_emails/sql_server_va_emails.pt:357]: Should this be `vaEmails` as `vaEmailList` field no longer exists?

```suggestion
    return db['vaEmails'] == '' || db['vaEmails'] == null || db['vaEmails] == undefined || db['vaEmails'].length == 0
```

The same with lines 379 and 399

File: security/azure/sql_server_va_emails/sql_server_va_emails.pt [hunk 7]
```diff
@@ -165,30 +408,50 @@ end
...
+      end
+      field "state" do
+        label "State"
+      end
+      field "version" do
+        label "Version"
+      end
+      field "service" do
+        label "Service"
+      end
```
Comment (nia-vf1) [inline security/azure/sql_server_va_emails/sql_server_va_emails.pt:454]: Should "vaEmails" field also be here as part of the export fields?

Comment (XOmniverse) [reply security/azure/sql_server_va_emails/sql_server_va_emails.pt:454]: No, because it is always empty by definition if its in the results.

**Review by nia-vf1 — APPROVED** [review]
LGTM

**Pull request merged**